src/dialect/mod.rs

```diff
@@ -304,6 +304,11 @@ pub trait Dialect: Debug + Any {
         false
     }
 
+    /// Returns true if the dialect supports numbers containing underscores
+    fn supports_underscore_separator(&self) -> bool {
```

Comment:
```suggestion
    fn supports_numeric_literal_underscores(&self) -> bool {
```
something like this? thinking `underscore_operator` wouldn't be descriptive on its own